Chess/humanplayer.cpp

```diff
@@ -1,61 +1,60 @@
 #include "humanplayer.h"
-#include <iostream>
 
 
-
-HumanPlayer::HumanPlayer(Color color) : Player(color) {}
+HumanPlayer::HumanPlayer(Color playerColor) : Player(playerColor) {}
 
 
 bool HumanPlayer::getMove(Piece* &piece, int &diffRow, int &diffCol)
 {
-
     Board* board = Board::getInstance();
 
-    int row = -1, column = -1;
-    int kind = WM_LBUTTONDOWN;
-
-    waitForClick(kind, row, column);
-
-
-    position humanPiece =
-    {
-        convertCoordinateX(column),
-        convertCoordinateY(row)
-    };
+    position humanPosition = waitForClick();
 
-    piece = board->getPieceForPosition(humanPiece.row, humanPiece.column);
+    piece = board->getPieceForPosition(humanPosition.row, humanPosition.column);
 
     if(piece == NULL)
         return 0;
 
-    if(piece->getColor() != color)
+    if(piece->getColor() != playerColor)
         return 0;
 
+    position targetPosition = waitForClick();
 
-    waitForClick(kind, row, column);
-
-    position enemyPiece =
-    {
-        convertCoordinateX(column),
-        convertCoordinateY(row)
-    };
-
-    diffRow = enemyPiece.row - humanPiece.row;
-    diffCol = enemyPiece.column - humanPiece.column;
-
-    Piece * _enemyPiece = board->getPieceForPosition(enemyPiece.row, enemyPiece.column);
+    diffRow = targetPosition.row - humanPosition.row;
+    diffCol = targetPosition.column - humanPosition.column;
 
+    Piece * enemyPiece = board->getPieceForPosition(targetPosition.row, targetPosition.column);
 
-    if(_enemyPiece != NULL && _enemyPiece->getColor() == piece->getColor())
+    if(enemyPiece != NULL && enemyPiece->getColor() == piece->getColor())
         return 0;
 
     return 1;
 }
 
 
-void HumanPlayer::waitForClick(int kind, int &row, int &column)
+position HumanPlayer::waitForClick()
```

Comment:
n-ar trb aici sa asteptam doar click-ul?
poate putem face o functie separata care returneaza position convertToPosition sau ceva la stilu

Reply:
Super ok commentul.
Pe moment am redenumit din `waitForClick` in `getClick` ca sa se inteleaga ca nu doar asteptam
Un lucru foarte important la naming e ca o functie sa nu faca mai mult decat spune :D GG!

Reply:
`getPositionClicked` ramane